src/coreComponents/dataRepository/DataContext.hpp

```diff
@@ -0,0 +1,270 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file DataContext.hpp
+ */
+
+#ifndef GEOS_DATAREPOSITORY_DATACONTEXT_HPP_
+#define GEOS_DATAREPOSITORY_DATACONTEXT_HPP_
+
+#include "common/DataTypes.hpp"
+#include "common/Logger.hpp"
+#include "xmlWrapper.hpp"
+#include "common/Format.hpp"
+
+namespace geos
+{
+
+namespace dataRepository
+{
+
+class Group;
+class WrapperBase;
+
+/// DataContext is an abstract class storing contextual information on an object:
+/// - its line position in a file, if applicable, implementation in DataFileContext,
+/// - its location in the data hierarchy, implementation in GroupContext and WrapperContext.
+/// Typically, the target object contains an unique_ptr< DataContext > instance of this class.
+class DataContext
```

Comment:
So it's a `HierarchyContext`?

Reply:
The `DataContext` have the following children:
- `GroupContext` and `WrapperContext`, which could be qualified of `HierarchyContext`.
- `DataFileContext`, which aims to give an object its source file contextual information (file / line / offset). 

I'm changing the comment here as it can be confusing:
```
/// DataContext is an abstract class storing contextual information on an object:
/// - either its line position in a file, if applicable, implementation in DataFileContext,
/// - or its location in the data hierarchy, implementation in GroupContext and WrapperContext.
/// Typically, the target object contains an unique_ptr< DataContext > instance of this class.
class DataContext
```

Do you think I should rename `DataContext` to `HierarchyContext`?